nitransforms/tests/test_linear.py

```diff
@@ -328,10 +328,10 @@ def test_mulmat_operator(testdata_path):
     mat2 = from_matvec(np.eye(3), (4, 2, -1))
     aff = nitl.Affine(mat1, reference=ref)
 
-    composed = aff @ mat2
+    composed = aff @ nitl.Affine(mat2)
     assert composed.reference is None
-    assert composed == nitl.Affine(mat1.dot(mat2))
+    assert composed == nitl.Affine(mat2 @ mat1)
```

Comment:
Can you add comment on why we should expect `Affine(mat1) @ Affine(mat2) == Affine(mat2 @ mat1)`? This seems counterintuitive.